src/snowflake/connector/cursor.py

```diff
@@ -733,13 +751,17 @@ def execute(
             self._log_telemetry_job_data(
                 TelemetryField.TIME_CONSUME_FIRST_RESULT, time_consume_first_result
             )
-        logger.debug("sfqid: %s", self.sfqid)
 
-        logger.info("query execution done")
         if ret["success"]:
             logger.debug("SUCCESS")
             data = ret["data"]
 
+            if "resultIds" in data:
+                self._init_multi_statement_results(data)
```

Comment:
I think this is going to be more involved than moving the code up unfortunately. We are going to need to make sure the regex works for multiple alter commands in a multi-statement query, then parse through all of the groups. I also noticed we probably want to run the following code for async multi-statements as well.
```
if _exec_async:
    self.connection._async_sfqids[self._sfqid] = None
if _no_results:
    self._total_rowcount = (
        ret["data"]["total"]
        if "data" in ret and "total" in ret["data"]
        else -1
    )
    return data
```

Maybe it would be best to detect PUT/GET commands in the returned payload after all. Then we don't need to rely on `self.is_file_transfer` just before these two code blocks.